src/test/scala/outwatch/LifecycleHookSpec.scala

```diff
@@ -34,6 +36,61 @@ class LifecycleHookSpec extends UnitSpec with BeforeAndAfterEach {
 
   }
 
+  "Insertion hooks" should "be called correctly on merged nodes" in {
+    var switch = false
+    val sink = Sink.create((_: Element) => IO(switch = true))
+    var switch2 = false
+    val sink2 = Sink.create((_: Element) => IO(switch2 = true))
+
+    val node = div(insert --> sink)(insert --> sink2)
+
+    switch shouldBe false
+    switch2 shouldBe false
+
+    OutWatch.render("#app", node).unsafeRunSync()
+
+    switch shouldBe true
+    switch2 shouldBe true
+
+  }
+
+  "Destruction hooks" should "be called correctly on merged nodes" in {
+
+    var switch = false
+    val sink = Sink.create((_: Element) => IO(switch = true))
+    var switch2 = false
+    val sink2 = Sink.create((_: Element) => IO(switch2 = true))
+
+    val node = div(child <-- Observable.of(span(destroy --> sink)(destroy --> sink2), "Hasdasd"))
+
+    switch shouldBe false
+    switch2 shouldBe false
+
+    OutWatch.render("#app", node).unsafeRunSync()
+
+    switch shouldBe true
+    switch2 shouldBe true
+
+  }
+
+  "Update hooks" should "be called correctly on merged nodes" in {
+    var switch1 = false
+    val sink1 = Sink.create((_: (Element, Element)) => IO(switch1 = true))
+    var switch2 = false
+    val sink2 = Sink.create((_: (Element, Element)) => IO(switch2 = true))
+
+    val message = Subject[String]()
+    val node = div(child <-- message, update --> sink1)(update --> sink2)
+
+    OutWatch.render("#app", node).unsafeRunSync()
+    switch1 shouldBe false
+    switch2 shouldBe false
+
+    message.next("wursi")
+    switch1 shouldBe true
+    switch2 shouldBe true
+  }
+
   "Destruction hooks" should "be called correctly" in {
```

Comment:
move to other `Destruction hooks` and use `it should ...`
(same for Insertion and Update hooks)